### PR TITLE
SC-5537 - move consent to user object

### PR DESCRIPTION
**Pull request by CordlessWool**
# Description

## Links to Tickets or other pull requests
https://github.com/schul-cloud/schulcloud-server/pull/1110
https://ticketsystem.schul-cloud.org/browse/SC-5537
https://github.com/hpi-schul-cloud/schulcloud-client/pull/2074

## Changes
In backend the consent is moving to users instead a own model. So the consentStatus could be part of user object

## Datasecurity <sub><sup>details [on Confluence](https://docs.schul-cloud.org/x/2S3GBg)</sup></sub>
Consent model will removed in future and the consent informations will be found in the user model

## Deployment
Migrationscript needs to run, moving the current consent to the user. Old model need to be persist till next step of update

## Screenshots
image

## Approval for review

- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### Link to Definiton of Done

More and detailed information on the _definition of done_ can be found [on Confluence](https://docs.schul-cloud.org/pages/viewpage.action?pageId=92831762)

### CODE REVIEW

File: src/pages/administration/teachers/index.vue [hunk 1]
```diff
@@ -34,15 +34,15 @@
 			<template v-slot:datacolumn-createdAt="{ data }">
 				{{ dayjs(data).format("DD.MM.YYYY") }}
 			</template>
-			<template v-slot:datacolumn-consent="{ data }">
-				<span v-if="data && data.consentStatus === 'ok'">
+			<template v-slot:datacolumn-consentStatus="{ data }">
+				<span v-if="data === 'ok'">
```
Comment (adrianjost) [inline src/pages/administration/teachers/index.vue:38]: maybe rename data to status for better readability?
```suggestion
			<template v-slot:datacolumn-consentStatus="{ data: status }">
				<span v-if="status === 'ok'">
```

File: deploy/Dockerfile.develop [hunk 2]
```diff
@@ -0,0 +1,22 @@
+FROM node:lts as builder
```
Comment (adrianjost) [inline deploy/Dockerfile.develop:1]: It seems unintutive to me to store a develop file in the deploy directory. I would move it up into the root of the repo.

File: Dockerfile.develop [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+RUN git submodule init
+RUN git submodule update --remote
+WORKDIR '/app/src/legacy-client'
+RUN npm i
+RUN gulp
+WORKDIR '/app'
+
+#EXPOSE 3200
+
+CMD [ "yarn", "dev:nuxt" ]
```
Comment (Parnswir) [inline Dockerfile.develop:22]: please remove this from the PR

**Comment by CordlessWool** [comment]
@update screenshots

**Comment by CordlessWool** [comment]
@schul-cloud-bot update screenshots

**Pull request merged**